lib/Reader/readerir.cpp

```diff
@@ -887,6 +887,89 @@ void GenIR::copyStruct(Type *StructTy, Value *DestinationAddress,
         Alignment, IsVolatile);
 }
 
+void GenIR::copyStruct(CORINFO_CLASS_HANDLE Class, IRNode *Dst, IRNode *Src,
+                       ReaderAlignType Alignment, bool IsVolatile,
+                       bool IsUnchecked) {
+  if (classHasGCPointers(Class)) {
+    GCLayout *RuntimeGCInfo = getClassGCLayout(Class);
+    const uint32_t PointerSize = getPointerByteSize();
```

Comment:
We went to a fair amount of trouble to have LLVM types correctly describe the GC references in value types. Seems like we should try and use that here instead of parsing the runtime GC info, especially because you seem to rely on the LLVM type info a bit lower down anyways.